app/scripts/directives/nav.js

```diff
@@ -42,6 +43,15 @@ angular.module('openshiftConsole')
           }
           return "project/" + $scope.projectName + href;
         };
+
+        $scope.show = function(item) {
+          var isValid = !item.isValid || item.isValid();
```

Comment:
I didn't figure these will be complex/expensive checks but can definitely switch this up.